controllers/aiven_application/reconciler.go

```diff
@@ -92,6 +92,16 @@ func (r *AivenApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		"secret_name": application.Spec.SecretName,
 	})
 
+	applicationDeleted, err := r.HandleProtectedAndTimeLimited(ctx, application, logger)
+	if err != nil {
+		utils.LocalFail("HandleProtectedAndTimeLimited", &application, err, logger)
+		return fail(err)
+	}
+
+	if applicationDeleted {
```

Comment:
To avoid this drifting away from the above, maybe it should be written as an "else if" clause on the previous if. That would also make it obvious that it also deals with the return from `HandleProtectedAndTimeLimited`.